core/state_transition.go

```diff
@@ -269,6 +269,13 @@ func (st *stateTransition) buyGas() error {
 			mgval = mgval.Add(mgval, l1Cost)
 		}
 	}
+	var operatorCost *uint256.Int
+	if st.evm.Context.OperatorCostFunc != nil && !st.msg.SkipNonceChecks && !st.msg.SkipFromEOACheck {
+		operatorCost = st.evm.Context.OperatorCostFunc(new(big.Int).SetUint64(st.msg.GasLimit), st.evm.Context.Time)
+		if operatorCost != nil {
+			mgval = mgval.Add(mgval, operatorCost.ToBig())
+		}
+	}
```

Comment:
Slight preference to combine these branches. Could be:

```go
if !st.msg.SkipNonceChecks && !st.msg.SkipFromEOACheck {
    if st.evm.Context.L1CostFunc != nil {
        // ...
    }
    if st.evm.Context.OperatorCostFunc != nil {
        // ...
    }
}
```

Prevents the duplicate checks on the `st.msg` settings + keeps the OP fee logic a bit more encapsulated.

Reply:
Agreed with this - otherwise the "skip" checks are likely to get out of sync between the two fees in the future accidentally and introduce bugs.

Reply:
Done